src/cpp/desktop/DesktopWebPage.cpp

```diff
@@ -403,7 +403,8 @@ bool WebPage::acceptNavigationRequest(const QUrl &url,
    
    // determine if this is a local request (handle internally only if local)
    std::string host = url.host().toStdString();
-   bool isLocal = host == "localhost" || host == "127.0.0.1" || host == "::1";
+   bool isLocal = host == "localhost" || host == "127.0.0.1" || host == "::1" ||
+                  url.scheme() == QStringLiteral("data");
```

Comment:
Is there any worry that this could be abused in some way to force RStudio to load some arbitrary data URI?

Reply:
No, the goal of this test is to make sure we're loading only local resources. By definition a data URI is a local resource, so it's at a similar trust level to localhost and loopback URIs. (In fact you could argue that the latter are less trustworthy since they could invoke an arbitrary endpoint handler on a local server and cause code to run, whereas a data URI can only be used to refer to static content.)